src/spec-node/collectionCommonUtils/publishCommandImpl.ts

```diff
@@ -39,7 +39,7 @@ export function getSemanticVersions(version: string, publishedVersions: string[]
 	return semanticVersions;
 }
 
-export async function doPublishCommand(version: string, ociRef: OCIRef, outputDir: string, output: Log, collectionType: string) {
+export async function doPublishCommand(version: string, ociRef: OCIRef, outputDir: string, output: Log, collectionType: string, archiveId?: string) {
```

Comment:
Maybe we pre-compute the result of `getArchiveName()`, and pass that as an argument to `doPublishCommand`?  Then, we wouldn't need an optional parameter that changes the previous behavior.

Reply:
nice 💡